### PR TITLE
Add VCH column to Containers section Datagrid

### LINKED ISSUE #5249: Add a VCH column to the containers tab in the H5 UI
The current implementation of the VIC H5 UI has a tab for containers. We should add a column to list the VCH that each container belongs to. Please see the mockup below

https://vmware.invisionapp.com/share/DA9NSB8CE#/screens/210384819

### PR DESCRIPTION
**Pull request by jooskim**
This PR adds the VCH column to the Datagrid showing all Container VMs in the H5C's VIC section.

image

Fixes #5249

### CODE REVIEW

File: ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java [hunk 1]
```diff
@@ -192,44 +218,88 @@ public ResultItem getVicVms(boolean isVch) {
...
+                        // process ResourcePool to populate vcValueRpNameMap
+                        vcValueRpNameMap = getVcMorValueRpNameMap(
+                                dpList,
+                                vcValueRpNameMap);
+                        continue;
+
+                    } else if (objType.equals(VsphereObjects.VirtualMachine)) {
+                        // process VirtualMachine by
+                        // 1. looking for config.guestFullName to determine
+                        //    if the VM is the desired VIC VM
```
Comment (anchal-agrawal) [inline ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java]: Minor: comment formatting should be consistent.

File: ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java [hunk 2]
```diff
@@ -192,44 +218,88 @@ public ResultItem getVicVms(boolean isVch) {
...
                                 }
                             }
-                            break;
                         }
                     }
+
+                    // if this ObjectContent is indeed either a VCH VM or a
+                    // Container VM then create an instance of its class and
+                    // add it to the array list
+                    if (isVicVm) {
```
Comment (anchal-agrawal) [inline ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java]: I think we can get rid of redundancy here with

```java
if (isVicVm) {
    PropertyValue pv = new PropertyValue();
    pv.propertyName = VsphereObjects.VmPropertyValueKey;
    if (isVch) {
        pv.value = new VirtualContainerHostVm(
            objC, serviceGuid);
    } else {
        pv.value = new ContainerVm(
            objC, serviceGuid);
    }
    pvList.add(pv);
    continue;
}
```

File: ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java [hunk 3]
```diff
@@ -242,6 +307,36 @@ public ResultItem getVicVms(boolean isVch) {
...
+	 * return a map containing VM ManagedObjectReference's "value" to
+	 * its parent object's name information 
+	 * @param dpList : List<DynamicProperty>
+	 * @param map : HashMap<String, String> to contain mapping info 
+	 * @return populated map with the provided dpList
+	 */
+	private Map<String, String> getVcMorValueRpNameMap(
+	        List<DynamicProperty> dpList,
+	        Map<String, String> map) {
+	    String resourcePoolName = null;
```
Comment (anchal-agrawal) [inline ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java]: Extra indent?

Comment (jooskim) [inline ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java]: I'd probably prefer to keep the content having 4-space indents which came right from line 317

Comment (jooskim) [inline ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/PropFetcher.java]: Thanks anyway :)

**Pull request merged**